cfme/tests/test_replication.py

```diff
@@ -349,7 +364,18 @@ def test_replication_global_to_remote_new_vm_from_template():
             2.
             3. VM created in the Remote, no errors.
     """
-    pass
+    remote_app, global_app = setup_replication
+    remote_provider = provider_app_crud(RHEVMProvider, remote_app)
+    remote_provider.setup()
+    assert remote_provider.name in remote_app.managed_provider_names, "Provider is not available."
+
+    new_vm_name = fauxfactory.gen_alphanumeric(start="test_replication_", length=25).lower()
+    global_provider = provider_app_crud(RHEVMProvider, global_app)
+    vm = create_vm(provider=global_provider, vm_name=new_vm_name)
+    remote_provider.refresh_provider_relationships()
+    assert new_vm_name in [vm.name for vm in remote_app.collections.infra_vms.filter
+    ({"provider": remote_provider}).all()], f"{new_vm_name} vm is not found in Remote Appliance"
+    request.addfinalizer(vm.cleanup_on_provider)
```

Comment:
Since this occurs after the `assert` this cleanup will not be executed. Finalizers need to be registered before the final assert of a test case. So we should move it to L375.